src/content/interop/objective-c-interop.md

```diff
@@ -51,144 +52,184 @@ and then load the library with `dart:ffi`.
 $ dart pub add --dev ffigen
 ```
 
-Then, configure ffigen to generate bindings for the
+Then, configure FFIgen to generate bindings for the
 Objective-C header containing the API.
-The ffigen configuration options go in your `pubspec.yaml` file,
-under a top-level `ffigen` entry.
-Alternatively, you can put the ffigen config in its own `.yaml` file.
+FFIgen can be configured via YAML or Dart code,
+but Dart is preferred for new projects.
+The YAML config will be deprecated in future versions of FFIgen.
+Start by creating a `generate_code.dart` script somewhere in your package.
+A good directory to create this file in is `my_package/tool`.
 
-```yaml
-ffigen:
-  name: AVFAudio
-  description: Bindings for AVFAudio.
-  language: objc
-  output: 'avf_audio_bindings.dart'
-  headers:
-    entry-points:
-      - '/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks/AVFAudio.framework/Headers/AVAudioPlayer.h'
-```
+The `generate_code.dart` script should create an `FfiGenerator` object,
+which will contain all our configuration options,
+then call its `.generate()` method.
+
+```dart
+import 'package:ffigen/ffigen.dart';
 
-The `name` is the name of the native library wrapper class
-that will be generated,
-and the `description` will be used in the documentation for that class.
-The `output` is the path of the Dart file that ffigen will create.
-The entry point is the header file containing the API.
-In this example, it is the internal `AVAudioPlayer.h` header.
-
-Another import thing you'll see,
-if you look at the [example config]({{page.example}}/pubspec.yaml),
-is the exclude and include options.
-By default, `ffigen` generates bindings for everything
-it finds in the header,
-and everything that those bindings depend on in other headers.
-Most Objective-C libraries depend on Apple's internal libraries,
-which are very large.
-If bindings are generated without any filters,
-the resulting file can be millions of lines long.
-To solve this problem,
-the ffigen config has fields that enable you to filter out
-all the functions, structs, enums, etc., that you're not interested in.
-For this example, we're only interested in `AVAudioPlayer`,
-so you can exclude everything else:
+final config = FfiGenerator(
+);
 
-```yaml
-  exclude-all-by-default: true
-  objc-interfaces:
-    include:
-      - 'AVAudioPlayer'
+void main() => config.generate();
 ```
 
-Since `AVAudioPlayer` is explicitly included like this,
-`ffigen` excludes all other interfaces.
-The `exclude-all-by-default` flag tells `ffigen` to
-exclude everything else.
-The result is that nothing is included except `AVAudioPlayer`,
-and its dependencies, such as `NSObject` and `NSString`.
-So instead of several million lines of bindings,
-you end up with tens of thousands.
+First, we'll tell FFIgen where to find the API we're trying to
+generate bindings for.
+To do this, we set the `headers.entryPoints` option.
+
+For this example, we need to load `AVAudioPlayer.h`.
```

Comment:
Generally the style guide suggests [addressing the reader as _you_](https://developers.google.com/style/person#address-the-reader-as-you):

```suggestion
First, you'll tell FFIgen where to find the API we're trying to
generate bindings for.
To do this, you set the `headers.entryPoints` option.

For this example, you'll load `AVAudioPlayer.h`.
```